lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -178,8 +186,7 @@ public void Close(byte[] message)
             }
             catch (Exception e)
             {
-                Utilities.Diagnostics.Exceptions.Caught(e, Level.Error, "Error during Close.", Logger);
-                _currentStatus.SetException(e);
+                Utilities.Diagnostics.Exceptions.CaughtAndThrow(e, Level.Error, "Error during Close.", Logger);
```

Comment:
`CaughtAndThrow` is misleading here, wording wise. We let the exception escape, right? Maybe this should be replaced with

```
   throw new TaskCloseException(TaskId, e);
```